PublishParasoftTestResults/xsl/cobertura.xsl

```diff
@@ -13,6 +14,118 @@
     </xsl:template>
 
     <xsl:template name="packages">
-        <xsl:element name="packages"></xsl:element>
+        <xsl:element name="packages">
+            <xsl:for-each-group select="/Coverage/Locations/Loc" group-by="substring-before(@uri, tokenize(@uri, '/')[last()])">
+                <xsl:element name="package">
+                    <xsl:attribute name="name">
```

Comment:
Instead of calling `callPackageNameTemplate `both inside `addPackageNameAttr  `and inside `addClassNameAttr`, let's only generate it once here and save as variable. Then use it in the two places.